src/library/autodj/autodjprocessor.cpp

```diff
@@ -180,33 +160,30 @@ AutoDJProcessor::AutoDJProcessor(
             ConfigKey(kConfigKey, kTransitionModePreferenceName), TransitionMode::FullIntroOutro);
 }
 
-AutoDJProcessor::~AutoDJProcessor() {
-    qDeleteAll(m_decks);
-    m_decks.clear();
-    delete m_pCOCrossfader;
-    delete m_pCOCrossfaderReverse;
-
-    delete m_pSkipNext;
-    delete m_pAddRandomTrack;
-    delete m_pShufflePlaylist;
-    delete m_pEnabledAutoDJ;
-    delete m_pFadeNow;
-
-    delete m_pAutoDJTableModel;
+void AutoDJProcessor::slotNumberOfDecksChanged(int decks) {
+    m_decks.reserve(decks);
```

Comment:
If I understand correctly, this slots could be called every time the number of decks changes, so you will needs to either reset the `m_decks` vector, or only add (or remove) the missing decks based on the new requested number.

Or is this slot not used like so?

Reply:
Mixxx does not support deleting decks. I will dorp a comment.

Reply:
Could we add a verify/assert there in case the deck count is being reduced? The CO can be decreased without any warnings or preventions (tested on Mixxx for Android as I am AFK)